### ISSUE TITLE
Positioning errors with vertical screen layout

### ISSUE THREAD

**Issue by delcake** [SnoutBug/mmcklauncher#23]
I run three monitors in my configuration, with the larger of the three placed above the other two and marked as the primary monitor. While trying out MMcK Launcher, it seems that it doesn't appear next to the launcher icon it was invoked from regardless of the placement setting specified in the configuration except for when the launcher is invoked from the top-most screen.

Example screenshots follow of what happens when I open the launcher from my bottom-left monitor.

Launcher Positioning Config: Default
![default_position](https://user-images.githubusercontent.com/228874/164609200-0ce954a5-f3f6-46dd-852a-5d37f4baf54e.png)

Launcher Positioning Config: Horizontal Center
![horizontal_center](https://user-images.githubusercontent.com/228874/164609317-8e0b7e7a-f275-4736-b779-d18e740b6da7.png)

Launcher Positioning Config: Screen Center
![screen_center](https://user-images.githubusercontent.com/228874/164609396-118ee08f-b758-40a4-8fed-16b23c1f5403.png)

Of the three, Screen Center is closest to correct. But I'm not able to tweak the position further to bring it down above the panel it was invoked from. The Horizontal Center option appears to have missed the mark the most by appearing on a totally different screen. Changing either of the two lower monitors to be the primary doesn't appear to change the behavior shown above at all.

Is there a way to ensure that the launcher always appears next to the launcher button and panel that it is opened from?

**Comment by SnoutBug** [comment]
I see, I will have a look once I got some more freetime. How does Menu 11 behave in your scenario? I've based most of my code on their launcher, so this might give me a clue as to what's missing / what I removed that broke it.

**Comment by delcake** [comment]
Menu 11 seems to be exhibiting similar behavior, regardless of how the Menu Positioning option is set. Testing once again by invoking from the bottom left monitor:

![Menu11](https://user-images.githubusercontent.com/228874/164764447-b179c940-962d-43ed-ab0f-b37dcc548794.png)

The exact position it chose seemed a bit suspicious to me, and so I checked what position the launcher ended up at. It's displaying at 1,020px from the top of the screen, which combined with the 50px panels I have and accounting for the 10px spacing it does from the panel when working correctly means that it is positioning as if it were on a 1080p screen. Swapping the top monitor to 1080p confirms it.

![Menu11_1080p](https://user-images.githubusercontent.com/228874/164765321-356d05c6-6377-408d-a36a-49a61b641f7b.png)

So as a shot in the dark, it seems the combination of my specific monitor orientation and mismatched screen resolutions are breaking an assumption or calculation about the nature of the workspace dimensions. It's using the size of the screen it was called from to know how far down to place the menu, but isn't accounting for another screen being present above it and so it positions from (0,0) instead of from (0,[Top Monitor Height]).

**Comment by delcake** [comment]
Good news, with some trial and error I managed to make a tweak to `MenuRepresentation.qml` that solved this placement issue for me. It handles when the panel is docked at both the bottom and top of any given screen. 

```
$ diff MenuRepresentation_original.qml MenuRepresentation.qml 
117c117
<             y = parent.height + panelSvg.margins.bottom + offset;
---
>             y = screen.y + parent.height + panelSvg.margins.bottom + offset;
122c122
<             y = screen.height - parent.height - height - panelSvg.margins.top - offset;
---
>             y = screen.y + screen.height - parent.height - height - panelSvg.margins.top - offset;
```
Though I freely admit I don't know what I'm doing, so the addition of `screen.y` might not catch every situation and might break in others. But at least in the testing for my specific configuration I couldn't come up with a scenario that didn't result in the launcher showing up where I intended it to.

Just to have a visual reference, I'll go ahead and include some more screenshots. This time, one invoking the launcher from each monitor in turn to show they appear where desired.
![working_bottomleft](https://user-images.githubusercontent.com/228874/165215518-004c7b05-a0c6-4f27-bd10-517e4fb83574.png)

![working_bottomright](https://user-images.githubusercontent.com/228874/165215559-3ab2c91d-47f1-4457-82cb-e87c6d5c3f69.png)

![working_top](https://user-images.githubusercontent.com/228874/165215588-ed7642f6-20ef-43c9-8376-f0e6467e97cb.png)

So ultimately, even if my edit isn't the exact answer, I hope it helps and thanks for looking in to it!
